src/lib/util/QueryBuilder.js

```diff
@@ -12,27 +12,24 @@ class QueryBuilder {
 	 */
 
 	/**
-	 * @typedef {QueryBuilderDatatype} QueryBuilderOptions
+	 * @typedef {Object} QueryBuilderOptions
+	 * @property {Object<string, QueryBuilderDatatype>} datatypes The datatypes
 	 * @property {Function} [formatDatatype] The datatype formatter for the SQL database
 	 * @property {Function} [arrayResolver] The specialized resolver for array keys
 	 */
 
 	/**
 	 * @since 0.5.0
-	 * @param {Object<QueryBuilderDatatype>} datatypes The datatype to insert
 	 * @param {QueryBuilderOptions} [options = {}] The default options for all datatypes plus formatDatatype
 	 */
-	constructor(datatypes = {}, { array = () => 'TEXT', resolver = null, type = null, arrayResolver, formatDatatype } = {}) {
+	constructor(options = {}) {
+		const { datatypes, array, resolver, type, arrayResolver, formatDatatype } = mergeDefault(options, QUERYBUILDER);
```

Comment:
No, I meant flatten datatypes into the rest, and use the rest operator to extract them.

Should be:
```js
const { array, resolver, type, arrayResolver, formatDatatype, ...datatypes } = mergeDefault(options, QUERYBUILDER);
```
ofcourse with the flattening done in constants.DEFAULTS.QUERYBUILDER aswell. For logical code seperation, can include a `// datatypes section` in the constants file.